ArchiSteamFarm/Actions.cs

```diff
@@ -282,7 +282,13 @@ public sealed class Actions : IDisposable {
 					TradingScheduled = false;
 				}
 
-				HashSet<Steam.Asset> inventory = await Bot.ArchiWebHandler.GetInventory(Bot.SteamID, appID, contextID, tradable: true, wantedRealAppIDs: wantedRealAppIDs, unwantedRealAppIDs: unwantedRealAppIDs, wantedTypes: wantedTypes).ConfigureAwait(false);
+				HashSet<Steam.Asset> inventory;
+
+				try {
+					inventory = await Bot.ArchiWebHandler.GetInventoryEnumerable(Bot.SteamID, appID, contextID).Where(item => item.Tradable && (wantedRealAppIDs?.Contains(item.RealAppID)).GetValueOrDefault(true) && !(unwantedRealAppIDs?.Contains(item.RealAppID)).GetValueOrDefault(false) && (wantedTypes?.Contains(item.Type)).GetValueOrDefault(true)).ToHashSetAsync().ConfigureAwait(false);
+				} catch (Exception e) {
+					return (false, string.Format(Strings.WarningFailedWithError, e.Message));
+				}
 
 				if ((inventory == null) || (inventory.Count == 0)) {
```

Comment:
When you mark the enumerable method as not null, Rider should signalize you that this inventory check against null is pointless, so just remove it (keep one for count).

Reply:
There might be more, check IDE.